tools/cpp/src/cpp-build/generate_geocoding_data.cc

```diff
@@ -102,20 +103,18 @@ bool ListDirectory(const string& path, vector<DirEntry>* entries) {
     return false;
   }
   AutoCloser<DIR> dir_closer(&dir, closedir);
-  struct dirent entry, *dir_result;
+  struct dirent *entry;
   struct stat entry_stat;
   while (true) {
-    const int res = readdir_r(dir, &entry, &dir_result);
-    if (res) {
-      return false;
-    }
-    if (dir_result == NULL) {
-      return true;
+    errno = 0;
```

Comment:
Given Fredrik's comment that this is slightly unobvious behaviour, can you add a comment explaining this here.
Setting errno explicitly to success before a call is not something I'm used to seeing.

Reply:
Fredrik's comment was on an outdated version of the file, I changed it. This is what he had suggested and if IIUC he doesn't think this one's unobvious - what he didn't think I should do was save the previous value of errno and set back to it if I come back with 0, which is what I had at commit 53642bf

Maybe let's discuss this in person. I'm happy with implementing it any way since you both know more here, but I'd like to understand it.